specification/subscription/resource-manager/Microsoft.Subscription/preview/2021-10-01-preview/subscriptions.json

```diff
@@ -0,0 +1,1156 @@
+          "type": "boolean"
+        },
+        "blockSubscriptionsIntoTenant": {
+          "description": "Blocks the entering of subscriptions into user's tenant.",
+          "type": "boolean"
+        },
+        "exemptedPrincipals": {
+          "description": "List of user objectIds that are exempted from the set subscription tenant policies for the user's tenant.",
+          "type": "array",
+          "items": {
+            "format": "uuid",
+            "type": "string",
+            "example": "00000000-0000-0000-0000-000000000000"
+          }
+        }
+      }
+    },
+    "Workload": {
+      "description": "The workload type of the subscription. It can be either Production or DevTest.",
+      "type": "string",
+      "x-ms-enum": {
+        "name": "workload",
+        "modelAsString": true
+      },
+      "enum": [
+        "Production",
+        "DevTest"
+      ]
+    },
+    "BillingScope": {
+      "description": "Billing scope of the subscription.\nFor CustomerLed and FieldLed - /billingAccounts/{billingAccountName}/billingProfiles/{billingProfileName}/invoiceSections/{invoiceSectionName}\nFor PartnerLed - /billingAccounts/{billingAccountName}/customers/{customerName}\nFor Legacy EA - /billingAccounts/{billingAccountName}/enrollmentAccounts/{enrollmentAccountName}",
+      "type": "string"
+    },
+    "AcceptOwnershipState": {
+      "readOnly": true,
+      "type": "string",
+      "description": "The accept ownership state of the resource.",
+      "enum": [
+        "Pending",
+        "Completed",
+        "Expired"
+      ],
+      "x-ms-enum": {
+        "name": "AcceptOwnership",
+        "modelAsString": true
+      }
+    },
+    "ProvisioningState": {
+      "readOnly": true,
+      "type": "string",
+      "description": "The provisioning state of the resource.",
+      "enum": [
+        "Pending",
+        "Accepted",
+        "Succeeded"
+      ],
+      "x-ms-enum": {
+        "name": "Provisioning",
+        "modelAsString": true
+      }
+    },
+    "ServiceTenantResponse": {
+      "description": "Billing account service tenant.",
+      "type": "object",
+      "properties": {
+        "tenantId": {
+          "description": "Service tenant id.",
+          "type": "string"
+        },
+        "tenantName": {
+          "description": "Service tenant name.",
+          "type": "string"
+        }
+      }
+    }
+  },
+  "parameters": {
+    "apiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "Version of the API to be used with the client request. Current version is 2021-10-01-preview"
+    },
+    "subscriptionIdParameter": {
+      "name": "subscriptionId",
+      "x-ms-parameter-location": "method",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Subscription Id."
+    },
```

Comment:
Instead of defining these yourself, please use $ref to the ones in common-types.

[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json)


[CADL/TypeSpec would auotatically do this for you in most cases, as an example of why its good to use]